source/extensions/filters/http/common/aws/credentials_provider.h

```diff
@@ -13,17 +13,27 @@ namespace HttpFilters {
 namespace Common {
 namespace Aws {
 
+/**
+ * AWS credentials container
+ *
+ * If a credential component was not found in the execution environment, it's getter method will
+ * return absl::nullopt. Credential components with the empty string value are treated as not found.
+ */
 class Credentials {
 public:
-  Credentials() = default;
-
-  Credentials(absl::string_view access_key_id, absl::string_view secret_access_key)
-      : access_key_id_(access_key_id), secret_access_key_(secret_access_key) {}
-
-  Credentials(absl::string_view access_key_id, absl::string_view secret_access_key,
-              absl::string_view session_token)
-      : access_key_id_(access_key_id), secret_access_key_(secret_access_key),
-        session_token_(session_token) {}
+  Credentials(absl::string_view access_key_id = absl::string_view(),
```

Comment:
Defaulting to `absl::string_view()` seems unnecessary? For convenience you always have the default constructor `Credentials()`?

Reply:
As I already have this constructor, I wouldn't get the default one unless I request it with `Credentials() = default`. To me it looks simpler to have a single constructor, because relying on the default one would not lead to code reduction.